Custom images list implementation

AndriiMysko opened the pull request.

DominikAlberski commented on lib/travis/api/v3/models/custom_image.rb:
```diff
@@ -0,0 +1,20 @@
+    scope :available, -> { where(state: 'available') }
+
+    def created_by
+      user_id = custom_image_logs.where(action: 'created').first&.sender_id
```
I think `find_by(action: 'created')&.sender_id` would be better. 

I'm not 100% sure, but if enumerator is used for the action column, you can do 'custom_image_logs.created.frist&.sender_id'

DominikAlberski commented on lib/travis/api/v3/models/custom_image_log.rb:
```diff
@@ -0,0 +1,6 @@
+module Travis::API::V3
+  class Models::CustomImageLog < Model
+    after_initialize :readonly!
+    belongs_to :custom_image
```
here you would need to add `enum action: {created: 'created', [etc.]}`

DominikAlberski reviewed (commented):
I would love to see some basic specs.

AndriiMysko commented:
> I would love to see some basic specs.

@DominikAlberski Yep, sorry, specs were added in the US4 PR for both 3 and 4 https://github.com/travis-ci/travis-api/pull/1369
If it's insufficient, please let me know. I've also addressed review comments.
